src/libraries/System.Private.CoreLib/src/System/Text/Latin1Utility.cs

```diff
@@ -0,0 +1,1101 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
+using Internal.Runtime.CompilerServices;
+
+#pragma warning disable SA1121 // explicitly using type aliases instead of built-in types
+#if TARGET_64BIT
+using nint = System.Int64;
+using nuint = System.UInt64;
+#else // TARGET_64BIT
+using nint = System.Int32;
+using nuint = System.UInt32;
+#endif // TARGET_64BIT
+
+namespace System.Text
+{
+    internal static partial class Latin1Utility
+    {
+#if DEBUG
+        static Latin1Utility()
+        {
+            Debug.Assert(sizeof(nint) == IntPtr.Size && nint.MinValue < 0, "nint is defined incorrectly.");
+            Debug.Assert(sizeof(nuint) == IntPtr.Size && nuint.MinValue == 0, "nuint is defined incorrectly.");
+        }
+#endif // DEBUG
+
+        /// <summary>
+        /// Returns the index in <paramref name="pBuffer"/> where the first non-Latin1 char is found.
+        /// Returns <paramref name="bufferLength"/> if the buffer is empty or all-Latin1.
+        /// </summary>
+        /// <returns>A Latin-1 char is defined as 0x0000 - 0x00FF, inclusive.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static unsafe nuint GetIndexOfFirstNonLatin1Char(char* pBuffer, nuint bufferLength /* in chars */)
+        {
+            // If SSE2 is supported, use those specific intrinsics instead of the generic vectorized
+            // code below. This has two benefits: (a) we can take advantage of specific instructions like
+            // pmovmskb which we know are optimized, and (b) we can avoid downclocking the processor while
+            // this method is running.
+
+            return (Sse2.IsSupported)
+                ? GetIndexOfFirstNonLatin1Char_Sse2(pBuffer, bufferLength)
```

Comment:
Were these split due to method size?

Reply:
It was done to make the method more friendly to human readers. (`AsciiUtility` reads largely the same way.) Did I inadvertently defeat JIT or AOT optimizations?